scripts/add_brownfield.py

```diff
@@ -0,0 +1,183 @@
+    
+    n_p.links.index=np.where(n_p.links.index.str[-4:].isin(previous_timesteps)==False,
+                                  n_p.links.index + '-' + previous_timestep, 
+                                  n_p.links.index)
+    n.madd("Link", 
+            n_p.links.index,
+            bus0=n_p.links.bus0,
+            bus1=n_p.links.bus1,
+            bus2=n_p.links.bus2,
+            carrier=n_p.links.carrier,
+            p_nom=n_p.links.p_nom_opt,
+            marginal_cost=n_p.links.marginal_cost,
+            capital_cost=n_p.links.capital_cost,
+            efficiency=n_p.links.efficiency,
+            efficiency2=n_p.links.efficiency2)
+    
+    
+if __name__ == "__main__":
+    # Detect running outside of snakemake and mock snakemake for testing
+    if 'snakemake' not in globals():
+        from vresutils.snakemake import MockSnakemake
+        snakemake = MockSnakemake(
+            wildcards=dict(network='elec', simpl='', clusters='37', lv='1.0', 
+                           sector_opts='Co2L0-168H-T-H-B-I-solar3-dist1',
+                           co2_budget_name='go',
+                           planning_horizons='2050'),
+            input=dict(network='pypsa-eur-sec/results/test/prenetworks/{network}_s{simpl}_{clusters}_lv{lv}__{sector_opts}_{co2_budget_name}_{planning_horizons}.nc', 
+                       network_p='pypsa-eur-sec/results/test/postnetworks/{network}_s{simpl}_{clusters}_lv{lv}__{sector_opts}_{co2_budget_name}_2040.nc',                        
+                       costs='pypsa-eur-sec/data/costs/costs_{planning_horizons}.csv',
+                       cop_air_total="pypsa-eur-sec/resources/cop_air_total_{network}_s{simpl}_{clusters}.nc",
+                       cop_soil_total="pypsa-eur-sec/resources/cop_soil_total_{network}_s{simpl}_{clusters}.nc"),                   
+            output=['pypsa-eur-sec/results/test/prenetworks_bf/{network}_s{simpl}_{clusters}_lv{lv}__{sector_opts}_{planning_horizons}.nc']
+        )
+        import yaml
+        with open('config.yaml') as f:
+            snakemake.config = yaml.load(f)
+
+    print(snakemake.input.network_p)
+    logging.basicConfig(level=snakemake.config['logging_level'])
+
+    options = snakemake.config["sector"]
+    opts = snakemake.wildcards.sector_opts.split('-')
+    
+    year=snakemake.wildcards.planning_horizons
+
+    n = pypsa.Network(snakemake.input.network,
+                      override_component_attrs=override_component_attrs)
+  
+    n_p = pypsa.Network(snakemake.input.network_p,
+                      override_component_attrs=override_component_attrs)
+
+    add_brownfield(n,n_p)
+    
+    Nyears = n.snapshot_weightings.sum()/8760.
+       
+    costs = prepare_costs()   
+    
+    baseyear=snakemake.config['scenario']["planning_horizons"][0]
+    
+    add_power_capacities_installed_before_baseyear(n, year, baseyear, costs) # only the capacities with YearDecomissioning < year are added
```

Comment:
YearDecomissioning > year?

Reply:
Yep, corrected in https://github.com/PyPSA/pypsa-eur-sec/pull/63/commits/a4c22a29a43f63e52d879267a7f26a5d27b5faa5